src/core/counts.jl

```diff
@@ -37,12 +37,12 @@ and `c.dimlabels[i]` is the label of the `i`-th dimension.
 Both labels and outcomes are assigned automatically if not given.
 `c` itself can be manipulated and iterated over like its stored array.
 """
-struct Counts{T <: Integer, N, S} <: AbstractArray{T, N}
+struct Counts{T <: Integer, N, O <: Tuple, S} <: AbstractArray{T, N}
     # The frequency table.
     cts::AbstractArray{T, N}
 
     # Outcomes[i] has the same number of elements as `cts` along dimension `i`.
-    outcomes::Tuple{Vararg{T, N} where T <: AbstractVector} where N
+    outcomes::O
```

Comment:
In some comment you mentioned that we should enforce the outcomes to be a tuple of `AbstractVector`s. This change no longer enforces that. If we still want to enforce this criterion, then we need to check that the input is actually a tuple of `AbstractVector` and error otherwise

Reply:
Removing this type specification also means that we need to add an additional check that the number of elements in the tuple is `N`, and error otherwise

Reply:
yes, correct. We need to add the inner constructor again that checks the tuple and length. I don't know how you made this work in the first place, but using `where` in one of the fields of the type isn't correct syntax. it errors.

Reply:
> yes, correct. We need to add the inner constructor again that checks the tuple and length. I don't know how you made this work in the first place, but using where in one of the fields of the type isn't correct syntax. it errors.

I'm not sure what you mean by "it errors". The code compiled successfully and tests passed as it was before you made the change.

I'm also not sure what you mean by "isn't correct syntax", since it works. The following code means:

```julia
struct MyType
    x::Tuple{Vararg{T, N} where T <: AbstractVector}
end
```

"`x` is an `N`-tuple where the element type is some subtype of `AbstractVector` (and is allowed to vary between the elements; hence the inner `where T <: AbstractVector`). Isn't this what we want?

Reply:
Ah, I see what you mean!

The last "`where N`" in `::Tuple{Vararg{T, N} where T <: AbstractVector} where N` shouldn't be there. I'm not sure how the tests were able to pass. Strange. Anyways, I'm hard-coding that this should be an `N`-tuple again, so API type correctness is always enforced